### DISCUSSION TITLE
Allow extending modifier builder with custom to*

### DISCUSSION THREAD

**Discussion by ImanMahmoudinasab** [evan-liu/karabiner.ts#138] [Q&A] [answered]
Thank you for this awesome work.
I think letting us define custom `to*` methods is a must for people like me with many reused scripts:
```ts
    map("l")
      .toChromeTab("http://localhost:8080")
      .description("Launch Localhost"),
```

**Comment by evan-liu** [comment, marked as answer]
You can define functions then use in `to()`, e.g.

```ts
map("l")
      .to(chromeTab("http://localhost:8080"))
      .description("Launch Localhost"),

function chromeTab(url) {
  return to$(`...`)
}
```

I have a few examples [in my config](https://github.com/evan-liu/karabiner-config/blob/f014529dad4a58bba05d7f4b39a90da14c483799/utils.ts#L94-L100)

**Reply by ImanMahmoudinasab** [reply]
Yeah this is what I end up doing. Having a consistent way of defining custom `to`s will be a nice to have feature, though.
